syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package autofill;

import "password_requirements.proto";

// A shard of password requirements for domains that can be served to Chrome
// clients.
message PasswordRequirementsShard {
  // A map that uses a domain suffix as a key and a PasswordRequirementsSpec as
  // a value.
  //
  // A key of "example.com" would mean the spec applies to "example.com",
  // "www.example.com" but not "some-example.com".
  //
  // It is valid to have one entry with key "www.example.com" and another entry
  // with key "example.com". In this case the most specific entry is applied by
  // Chrome.
  map<string, PasswordRequirementsSpec> specs = 1;
}
